azurelinuxagent/ga/update.py

```diff
@@ -314,10 +315,18 @@ def run(self, debug=False):
 
             #
             # Initialize the goal state; some components depend on information provided by the goal state and this
-            # call ensures the required info is initialized (e.g telemetry depends on the container ID.)
+            # call ensures the required info is initialized (e.g. telemetry depends on the container ID.)
             #
             protocol = self.protocol_util.get_protocol()
```

Comment:
The `get_protocol` function has its own set of retries (360 retries with 10 sec sleep between each retry). Based on that logic, I chose not to add any more retries here.

Reply:
invalid comment, not relevant anymore